src/nsrt_learning/option_learning.py

```diff
@@ -412,31 +416,29 @@ def learn_option_specs(self, strips_ops: List[STRIPSOperator],
             # We do this for learning efficiency; including all objects would
             # likely work too, but may require more data for the model to
             # realize that those objects' parameters can be ignored.
-            changing_parameter_set = self._get_changing_parameters(datastore)
+            # Furthermore, we only want to include the features of the objects
+            # that exhibit some change in the data.
+            changing_var_to_feat = self._get_changing_features(datastore)
             # Just to avoid confusion, we will insist that the order of the
             # changing parameters is consistent with the order of the original.
-            changing_parameters = sorted(changing_parameter_set,
-                                         key=op.parameters.index)
-            del changing_parameter_set  # not used after this
-            assert changing_parameters == [
-                v for v in op.parameters if v in changing_parameters
-            ]
-
+            changing_var_order = sorted(changing_var_to_feat,
+                                        key=op.parameters.index)
             for segment, var_to_obj in datastore:
                 all_objects_in_operator = [
                     var_to_obj[v] for v in op.parameters
                 ]
 
                 # First, determine the absolute goal vector for this segment.
```

Comment:
is this comment up to date? i don't understand what here is the "absolute goal vector". i feel like this chunk is actually computing the option parameters.

Reply:
oh, is final_params the absolute goal vector? can we just clarify this comment significantly, to state that we're accomplishing two things here: (1) computing option_param so they can later be stored into the segments, (2) computing final_param which is the absolute goal vector so we can later compute relative goal vectors when we iterate through the segment's states